lib/srv/sess.go

```diff
@@ -211,6 +211,25 @@ func (s *SessionRegistry) OpenSession(ch ssh.Channel, req *ssh.Request, ctx *Ser
 
 		return nil
 	}
+
+	if ctx.srv.GetCreateHostUser() &&
+		ctx.Identity.RoleSet.CreateHostUser(s.srv.GetInfo()) {
+		mgmt, err := NewUserManagement()
```

Comment:
Can "user management" service be a part of the SessionRegistry instead of being instantiated every time? Call it e.g. `Users`, then you'd do something like `s.Users.Create(xxx)`.